### PR TITLE
[resource/host] Define host.mac semantic convention

### PR DESCRIPTION
**Pull request by mx-psi**
Updates #131

## Changes

Add `host.mac` semantic convention to list MAC addresses of a host.

## Merge requirement checklist

* [ ] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [ ] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [ ] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/resource/host.md [hunk 1]
```diff
@@ -16,6 +16,9 @@
 | `host.image.name` | string | Name of the VM image or OS install the host was instantiated from. | `infra-ami-eks-worker-node-7d4ec78312`; `CentOS-8-x86_64-1905` | Recommended |
 | `host.image.id` | string | VM image ID or host OS image ID. For Cloud, this value is from the provider. | `ami-07b06b442921831e5` | Recommended |
 | `host.image.version` | string | The version string of the VM image or host OS as defined in [Version Attributes](README.md#version-attributes). | `0.1` | Recommended |
+| `host.mac` | string[] | Available MAC addresses of the host, excluding loopback interfaces. [1] | `[ac-de-48-23-45-67, AC-DE-48-23-45-67-01-9F]` | Opt-In |
+
+**[1]:** MAC Addresses MUST be represented in [IEEE RA hexadecimal form](https://standards.ieee.org/wp-content/uploads/import/documents/tutorials/eui.pdf): as hyphen-separated octets in hexadecimal form from most to least significant.
```
Comment (mx-psi) [inline docs/resource/host.md:21]: Not sure if this should be a MUST; hyphens seems to be the IEEE standard but I don't think it is unusual to see MAC addresses separated by colons (e.g. Go will separate by `:`, see https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/net/mac.go;l=19). Maybe it should be a SHOULD instead? Or maybe we should abandon IEEE RA notation and do "IEEE RA notation but with colons"?

Comment (andrewkroh) [reply docs/resource/host.md:21]: There was a similar discussion for ECS at https://github.com/elastic/ecs/pull/456#issuecomment-767099447. This comment mostly sums it up how it was decided.

> There are two competing standards. Lets just go with one of them. Better than having implementers choose one at random.

I think a stricter definition will be better for consistency, and this predictability is good if you are trying to correlate or search. Toward that goal of consistency in the data, I think the definition should advise either lowercase or uppercase.

(new here, so if this goes against some core principles please let me know, thanks)

Comment (mx-psi) [reply docs/resource/host.md:21]: The ECS conclusion sounds reasonable to me. I did not specify casing intentionally to give some leeway to implementations, but I guess it's better to be stricter; we can always relax the condition if it seems impractical in the future.

Comment (mx-psi) [reply docs/resource/host.md]: Changed, thanks for the suggestion and the context :)

**Review by AlexanderWert — APPROVED** [review]

**Review by ChrsMark — APPROVED** [review]

**Comment by mx-psi** [comment]
@open-telemetry/semconv-system-approvers PTAL!

File: model/resource/host.yaml [hunk 2]
```diff
@@ -79,6 +79,15 @@ groups:
           IPv4 Addresses MUST be specified in dotted-quad notation. IPv6 addresses
           MUST be specified in the [RFC 5952](https://www.rfc-editor.org/rfc/rfc5952.html) format.
         examples: ["192.168.1.140", "fe80::abc2:4a28:737a:609e"]
+      - id: mac
```
Comment (frzifus) [inline model/resource/host.yaml:82]: I think this is the same controversy as `host.ip`.
We are adding `host.mac` which represends multiple mac addresses to be consistent with ECS and we again violated the otel recommendations.

Comment (AlexanderWert) [reply model/resource/host.yaml:82]: For reference: https://github.com/open-telemetry/semantic-conventions/pull/333/files

**Review by dmitryax — APPROVED** [review]

File: model/resource/host.yaml (hunk 2 above)
Comment (mx-psi) [reply model/resource/host.yaml:82]: @frzifus what would you suggest doing here to move forward?

Comment (frzifus) [reply model/resource/host.yaml:82]: @AlexanderWert I assume you reference to:
>   - Do not modify an ECS name for the only reason of complying with [OpenTelemetry general attribute naming guidelines](https://opentelemetry.io/docs/specs/otel/common/attribute-naming/).

I am fundamentally unsure whether this guideline is a good idea. However, I don't have a very strong opinion on it, as I am not very familiar with ECS.

That said, if everyone agrees, I don't want to block the change.

cc @mx-psi

**Comment by mx-psi** [comment]
@open-telemetry/specs-semconv-maintainers This is ready for a review from maintainers.

The only remaining point of contention is the name, since `host.mac` could be argued to be singular but its type is an array, similar to the discussion at #203. We don't have a strong opinion on the system semconv WG over whether the plural rule should be followed as opposed to preserving backwards compatibility with ECS, so we believe it is up to maintainers now to decide on this :)

**Comment by joaopgrassi** [comment]
This is essentially the same discussion as with the `host.ip` and ECS. I'm inclined to accept it under `host.mac` for the same reasons the other was accepted.

File: CHANGELOG.md [hunk 3]
```diff
@@ -45,6 +45,9 @@ release.
   ([#267](https://github.com/open-telemetry/opentelemetry-specification/pull/267))
 - Add opt-in `system.memory.limit` metric.
   ([#409](https://github.com/open-telemetry/semantic-conventions/pull/409))
+- Add `host.mac` resource attribute convention.
+  ([#340](https://github.com/open-telemetry/semantic-conventions/pull/340))
+
```
Comment (AlexanderWert) [inline CHANGELOG.md:50]:
```suggestion
```

**Review by jsuereth — APPROVED** [review]

**Pull request merged**